core/base/src/TError.cxx

```diff
@@ -56,12 +56,8 @@ asm(".desc ___crashreporter_info__, 0x10");
 
 static ErrorHandlerFunc_t gErrorHandler = DefaultErrorHandler;
 
-
-/// Mutex for error and error format protection
-static std::mutex &GetErrorMutex()
-{
-   static std::mutex m;
-   return m;
+namespace {
+std::mutex gErrorPrintMutex;
```

Comment:
Can you add a comment in the code explaning why the function-static version does not work?  This will avoid our future self from trying out this "improvement" again later on :)  Thanks.

Reply:
I somehow don't find a reply to https://github.com/root-project/root/pull/5762#issuecomment-637578021 Is it Github or is it not yet addressed?

Reply:
That's the question: so far I only found out that in this case, on macOS, a global static and a magic static do _not_ behave the same. It seems to work with a global static, and apparently the `std::mutex` constructor is `constexpr` and `noexcept`, so that should be fine.

How important is finding out the root cause of the magic static failure?

Reply:
All right, this seems to be the answer: https://stackoverflow.com/questions/49342421/is-stdmutex-trivially-destructible

On clang/macOS, std::mutex is not trivially destructible.  So `pthread_mutex_destroy` is called before all other statics are destroyed.  One of these other statics must use error printing during destruction and that triggers a SIGABORT on the invalid mutex.

Reply:
Now thinking about it the function static is indeed more fragile .. maybe.  For example we may have the following sequence of event:
```
1. libCore.so is loaded # a static mutex would be initilialized here.
2. OtherLibrary is loaded
3. Some Warning message is issued # a function static mutex would be intialized here.
3. end of main
4. Other libary is unloaded and tries to print a Warning message
5. libCore.so is unloaded # static mutex would destructed here.
```
It is unclear (to me, right now) if the function static would be destructed between (3) and (4) [crash] or between (4)  and (5) [good run].